api/client.go

```diff
@@ -89,6 +90,7 @@ type RepositoryClient interface {
 	Symlink(ctx context.Context, repoID, ref, path string) (string, error)
 
 	RefsDump(ctx context.Context, repository string) (*models.RefsDump, error)
+	RefsRestore(ctx context.Context, repository string, manifest *models.RefsDump) error
```

Comment:
Getting ever more worried about the size of `manifest`.  We should consider (*not in this PR!*) exporting it and passing around a reference.

Reply:
not sure I understand. the manifest is a tiny json file externally, and a small 3 field struct when deserialized. I'm probably missing your meaning though.

Reply:
Yet another overloaded word. If you have a concrete suggestion to rename it, i'll be OK with that :)